frontend/app/work_packages/services/work-package-field-service.js

```diff
@@ -91,8 +91,15 @@ module.exports = function(
     return isEmpty(workPackage, field);
   }
 
+  function isMilestone(workPackage) {
+    return workPackage.form.embedded.payload.links.type.props.href === '/api/v3/types/4';
```

Comment:
Now I see that there is also an outdated response by @0xF013 
> @Azure7111 we need to know that the type in the form response is a Milestone because we first do a post
> request to the form endpoint, and it returns only a link to the type. I am not sure if making another blocking
> request to the type endoint on each type change is worth it

I'd say you have no choice, because you can't know otherwise whether something is a milestone or not.

I don't know your code, so my advise here might or might not be helpful:

`workPackage.form.embedded.schema.type.embedded.allowedValues` should contain a list of fully resolved types, those types that you can currently choose.

I still want to point out an API restriction here (which is a restriction-by-specifiction, not a restriction-by-implementation):

Whenever there is something like a `_link` to a `type` or `allowedValues` and for the same link there is something `_embedded`, then you are only **guaranteed** to get the link, because the embedded stuff is a *shortcut* to avoid downloading the stuff in question. However, you are not **guaranteed** that the embedded part is present. The embedding is only an optimization to avoid that the client needs to resolve each link manually.